cli/src/pcluster/config/cluster_config.py

```diff
@@ -1173,6 +1176,15 @@ def _register_validators(self):
         if self.custom_s3_bucket:
             self._register_validator(S3BucketValidator, bucket=self.custom_s3_bucket)
             self._register_validator(S3BucketRegionValidator, bucket=self.custom_s3_bucket, region=self.region)
+        self._register_validator(SchedulerOsValidator, scheduler=self.scheduling.scheduler, os=self.image.os)
+        self._register_validator(
+            HeadNodeImdsValidator, imds_secured=self.head_node.imds.secured, scheduler=self.scheduling.scheduler
```

Comment:
When doing this kind of refactoring it's better to split the refactoring change from the new feature in two different commits.